tock/api/views.py

```diff
@@ -129,6 +130,41 @@ class TimecardSerializer(serializers.Serializer):
         allow_null=True
     )
 
+class FullTimecardSerializer(serializers.ModelSerializer):
+    # Fields that require accessing other models
+    user_name = serializers.CharField(source='user.username')
+    reporting_period_start_date = serializers.DateField(source='reporting_period.start_date')
+    reporting_period_end_date = serializers.DateField(source='reporting_period.end_date')
+    hours_spent = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Timecard
+        fields = [
+            # straight pass-through fields
+            'id',
+            'submitted',
+            'submitted_date',
+            'billable_expectation',
+            'target_hours',
+            'billable_hours',
+            'non_billable_hours',
+            'excluded_hours',
+            'utilization',
+            # fields from other models
+            'user_name',
+            'reporting_period_start_date',
+            'reporting_period_end_date',
+            # fields that require something more complicated (e.g. an aggregation)
+            'hours_spent'
+        ]
+
+    def get_hours_spent(self, obj):
```

Comment:
I added this before realizing that `Timecard` has a `calculate_hours` method that updates various `_hours` fields when a `Timecard` object is updated. Are those `_hours` fields sufficient, or do we want to include this `total_hours` field too?

Reply:
Great question! The fields on `Timecard` are sufficient, any extra calculations for these instances on the way out are unexpected and _shouldn't_ be needed

Reply:
Sounds good!